src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchSimplificationFixAllProvider.cs

```diff
@@ -59,8 +67,33 @@ protected virtual SyntaxNode GetNodeToSimplify(SyntaxNode root, SemanticModel mo
             }
 
             // Add simplifier and formatter annotations to all nodes to simplify.
-            root = root.ReplaceNodes(nodesToSimplify, (o, n) =>
-                n.WithAdditionalAnnotations(Simplifier.Annotation, Formatter.Annotation));
+            if (!NeedsParentFixup)
+            {
+                root = root.ReplaceNodes(nodesToSimplify, (o, n) =>
+                    n.WithAdditionalAnnotations(Simplifier.Annotation, Formatter.Annotation));
+            }
+            else
+            {
+                var annotation = new SyntaxAnnotation();
+                root = root.ReplaceNodes(nodesToSimplify, (o, n) =>
+                    o.WithAdditionalAnnotations(annotation));
+                document = document.WithSyntaxRoot(root);
+
+                while (true)
+                {
+                    root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var annotatedNodes = root.GetAnnotatedNodes(annotation);
+                    var annotatedNode = annotatedNodes.FirstOrDefault(n => !n.HasAnnotation(Simplifier.Annotation));
+                    if (annotatedNode == null)
+                    {
+                        root = root.ReplaceNodes(annotatedNodes, (o, n) => o.WithoutAnnotations(annotation));
+                        break;
+                    }
+
+                    document = await AddSimplifyAnnotationsAsync(document, annotatedNode, cancellationToken).ConfigureAwait(false);
```

Comment:
Can you please add a bunch of comments here as to why we need the parent fixups.